simplified-app-shared/src/main/java/org/nypl/simplified/app/AnnotationsManager.kt

```diff
@@ -0,0 +1,600 @@
+    val LOG = LoggerFactory.getLogger(AnnotationsManager::class.java)!!
+  }
+
+  /**
+   * Queries the server, and based on a set of conditions, decides whether
+   * or not to inform the user about "Syncing" in an Alert Dialog, where
+   * the user can choose to dismiss/ignore, or turn it on for the first time.
+   * @param account The particular library for the permission status request.
+   * @param completion Asynchronous handler for the user's decision, or error.
+   */
+  fun requestServerSyncPermissionStatus(account: AccountsControllerType,
+                                        completion: (enableSync: Boolean) -> Unit)
+  {
+    if (!syncIsPossible(account)) {
+      LOG.debug("Account does not satisfy conditions for sync setting request.")
+      completion(false)
+      return
+    }
+
+    //TODO I think this condition is either wrong or out of date.. think about it some more during testing
+    if (Simplified.getSharedPrefs().getBoolean("userHasSeenFirstTimeSyncMessage") == true &&
+        Simplified.getSharedPrefs().getBoolean("syncPermissionGranted", libraryAccount.id) == false) {
+      completion(false)
+      return
+    }
+
+    syncPermissionStatusUriRequest { initialized, syncIsPermitted ->
+      if (initialized && syncIsPermitted) {
+        Simplified.getSharedPrefs().putBoolean("userHasSeenFirstTimeSyncMessage", true)
+        LOG.debug("Sync has already been enabled on the server. Enable here as well.")
+        completion(true)
+      } else if (!initialized && !Simplified.getSharedPrefs().getBoolean("userHasSeenFirstTimeSyncMessage")) {
+        LOG.debug("Sync has never been initialized for the patron. Proceeding with opt-in alert.")
+        presentFirstTimeSyncAlertDialog(completion)
+      } else {
+        LOG.debug("Sync has been initialized and not permitted. Continuing with sync disabled.")
+        completion(false)
+      }
+    }
+  }
+
+  private fun syncPermissionStatusUriRequest(completion: (initialized: Boolean, syncIsPermitted: Boolean) -> Unit)
+  {
+    val catalogUriString = libraryAccount.catalogUrl ?: null
+    val baseUri = if (catalogUriString != null) Uri.parse(catalogUriString) else null
+    val permissionRequestUri = if (baseUri != null) Uri.withAppendedPath(baseUri, "patrons/me/") else null
+
+    if (permissionRequestUri == null) {
+      LOG.debug("Could not create Annotations URL from Main Feed URL. Abandoning attempt to retrieve sync setting.")
+      return
+    }
+
+    val request = NYPLJsonObjectRequest(
+        Request.Method.GET,
+        permissionRequestUri.toString(),
+        credentials.barcode.toString(),
+        credentials.pin.toString(),
+        null,
+        Listener<JSONObject> { response ->
+          handleSyncPermission(response, completion)
+        },
+        Response.ErrorListener { error ->
+          LOG.error("GET request fail! Error: ${error.message}")
+        })
+
+    request.retryPolicy = DefaultRetryPolicy(
+        60 * 1000,
```

Comment:
can we give these numbers names? `secondsInMinute * minutesUntilTimeout` or something

Reply:
Maybe even make this value a `DEFAULT_RETRY_TIMEOUT` const on `DefaultRetryPolicy` or something of the like

Reply:
I really like `TimeUnit.SECONDS.toMillis(60)` for these cases

Reply:
How about I take both of your advice?